syntax = "proto3";

package test;

import "profanedb/protobuf/options.proto";

import "nested.proto";

message Test {
    int32 field_one_int = 1 [ (profanedb.protobuf.options).key = true ];
    string field_two_str = 2;
    bool field_three_bool = 3;

    Nested field_five_nested = 5;
    bytes field_four_bytes = 4;

    ExternalNested field_six_externalnested = 6;

    Nested.KeyableNestedInNested field_seven_repeated_keyable = 7;
}

message Nested {
    string nested_field_one_str = 1 [ (profanedb.protobuf.options).key = true ];
    int64 nested_field_two_int = 2;
    double nested_field_three_double = 3;

    message KeyableNestedInNested {
        repeated string nested_in_nested_field_one_repeated_str = 1 [ (profanedb.protobuf.options).key = true ];
    }

//    message NotKeyableNestedInNested {
//        int32 nested_in_nested_field_one_int = 1;
//    }
}
